### ISSUE TITLE
wkhtmltopdf intermittently hanging

### ISSUE THREAD

**Issue by jaztech** [wkhtmltopdf/wkhtmltopdf#2113]
I am running wkhtmltopdf 0.12.2-dev-5dea253 (with patched qt) on a CentOS server release 6.6 (Santiago)

In **about 1 of every 10 runs** the process hangs while "Loading Pages" around 88-90%. The error occurs whether it is called from PHP or if run on the command line with the same command.

Loading pages (1/6) [=====================================================> ] 89% (hangs here and never returns)

I have tried different permission settings and different versions of wkhtmltopdf. The only time I was able to stop this error from occurring was with an older version of wkhtmltopdf, but the resulting PDF was not acceptable output.

Command I am running:
wkhtmltopdf -B 0mm -T 0mm -R 0mm -L 0mm -s Letter --dpi 100 --disable-smart-shrinking --print-media-type {html file} {pdf file}

**Comment by Wayne-Wen** [comment]
I'm having the same issue. Have you @jaztech found any solutions?

**Comment by ashkulz** [comment]
@Wayne-Wen: without something that can be reproduced, I can't help you out.

**Comment by Wayne-Wen** [comment]
@ashkulz The command I am running is:
/usr/local/bin/wkhtmltopdf --lowquality --zoom '0.77' --footer-font-size '7' --footer-right '[page]/[topage]' "http://nbs.net.au/info/confirmation?b=10010045&p=86be8f1d8d6af3a00d8288ebbfc03944" "/tmp/test1.pdf"
I am using ten terminal to run this command. I have set different out put file for different terminal. Sometimes, all of them rum successfully. Sometimes, one of them will stuck on loading page without any error.

**Comment by ashkulz** [comment]
Are you sure it is not a network issue? Try to reproduce it with a local file.

**Comment by Wayne-Wen** [comment]
If it is a network issue, why other command run successfully? Is there any way to let wkhtmltopdf to show which file cannot be loaded? When I tried, the "Loading pages" was hanging, looks like something is blocking.

**Comment by jaztech** [comment]
We narrowed the issue down to having something to do with pulling remote resources in for the PDF.  If we pulled the image and javascript files down locally before running wkhtmltopdf, the intermittent hanging stopped.  We couldn't find any errors on the network stack, it was strictly related to wkhtmltopdf.  We could only reproduce it when generating large PDFs.  When generating 10-20 page PDFs, the hanging never happened.  When generating 40+ page PDFs it happened about 1 out of 10 runs.

Our solution was to create a wrapper script around wkhtmltopdf that pulled all remote content down locally via wget before running wkhtmltopdf.  Haven't had any issues since, except for the time someone put a link in the HTML and wget tried to pull down the entire website the link pointed to before calling wkhtmltopdf (we now forbid end users from entering non-formatting tags into fields ;-)).
